src/node_contextify.cc

```diff
@@ -132,41 +133,51 @@ class ContextifyContext {
-        // this point.  ForceSet(key,val,att) only supports value properties
-        // with the ES3-style attribute flags (DontDelete/DontEnum/ReadOnly),
-        // which doesn't faithfully capture the full range of configurations
-        // that can be done using Object.defineProperty.
-        if (clone_property_method.IsEmpty()) {
-          Local<String> code = FIXED_ONE_BYTE_STRING(env()->isolate(),
-              "(function cloneProperty(source, key, target) {\n"
-              "  if (key === 'Proxy') return;\n"
-              "  try {\n"
-              "    var desc = Object.getOwnPropertyDescriptor(source, key);\n"
-              "    if (desc.value === source) desc.value = target;\n"
-              "    Object.defineProperty(target, key, desc);\n"
-              "  } catch (e) {\n"
-              "   // Catch sealed properties errors\n"
-              "  }\n"
-              "})");
-
-          Local<Script> script =
-              Script::Compile(context, code).ToLocalChecked();
-          clone_property_method = Local<Function>::Cast(script->Run());
-          CHECK(clone_property_method->IsFunction());
+        Local<Object> desc_from_vm_context = Local<Object>::Cast(global
+            ->GetOwnPropertyDescriptor(context, key).ToLocalChecked());
+
+        bool isAccessor = desc_from_vm_context
+            ->Has(context, env()->get_string())
+            .FromJust() || desc_from_vm_context
+                ->Has(context, env()->set_string())
+                .FromJust();
+
+        PropertyDescriptor* desc;
+
+        if (!isAccessor) {
+          Local<Value> value = Local<Value>::Cast(desc_from_vm_context
+              ->Get(context, env()->value_string())
+              .ToLocalChecked());
+
+          bool writable = Local<Value>::Cast(desc_from_vm_context
+              ->Get(context, env()->writable_string())
+              .ToLocalChecked())->BooleanValue();
+
+          desc = new PropertyDescriptor(value, writable);
+        } else {
+          Local<Function> get = Local<Function>::Cast(desc_from_vm_context
+              ->Get(context, env()->get_string())
+              .ToLocalChecked());
+          Local<Function> set = Local<Function>::Cast(desc_from_vm_context
+              ->Get(context, env()->set_string())
+              .ToLocalChecked());
+
+          desc = new PropertyDescriptor(get, set);
         }
-        Local<Value> args[] = { global, key, sandbox_obj };
-        clone_property_method->Call(global, arraysize(args), args);
-      }
+
+        desc->set_configurable(Local<Value>::Cast(desc_from_vm_context
+            ->Get(context, env()->configurable_string()).ToLocalChecked())
+            ->BooleanValue());
+        desc->set_enumerable(Local<Value>::Cast(desc_from_vm_context
+            ->Get(context, env()->enumerable_string())
+            .ToLocalChecked())->BooleanValue());
+
+        sandbox_obj->DefineProperty(context, key, *desc);
+
+        delete desc;
     }
   }
+}
```

Comment:
Thanks @TimothyGu.
I have just re-compiled it and re-run the tests. All works. Could you double check, please, that it fails?